### ISSUE TITLE
Assumed spack wouldn't run -j 20 on a login node

### ISSUE THREAD

**Issue by markcmiller86** [LLNL/spack#30317]
### Summary

spack version: 0.18.0.dev0 (b00983439fc3f11365664853d80cb7f497fd88a6)

1. clone spack repo on LLNL CZ system (like pascal)
2. set up compilers as normal
3. run `spack install visit%gcc@6.1.0 ^python+shared ^mesa+opengl`
4. look at load average and top output
5. You will see load average jump to over 10 and many spack build processes

It didn't occur to me to go look for this but I got worried I was overloading a login node and sure enough, the data suggests I was. But, maybe thats because Spack had `niced` all the jobs and there were no other users attempting to use the login node (interactively) at that time. 

Nonetheless, I got nervous and moved to a compute node with `sxterm`

Assuming spack was not being `nice` about all the processes it was launching, I was thinking it would be good for it to not run `-j 20` (or whatever) when its on a login node.

### Rationale

Yes. I think it would be good if spack was smart about taking up login node resources

### Description

If spack can easily detect its on a login node, automatically reduce simultaneous process count to say just 4, that might be a bit friendlier to such a precious resource. I always get bothered with others that wind up ramping up load average on a login node so much interactive operations (e.g. `vi`) are no longer interactie.

### General information

- [X] I have run `spack --version` and reported the version of Spack
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by mwkrentel** [comment]
@markcmiller86 Are you sure it was 20 and not 16?

This is controlled by `build_jobs` in `config.yaml`.

```
  # The maximum number of jobs to use for the build system (e.g. `make`), when
  # the -j flag is not given on the command line. Defaults to 16 when not set.
  # Note that the maximum number of jobs is limited by the number of cores
  # available, taking thread affinity into account when supported. For instance:
  # - With `build_jobs: 16` and 4 cores available `spack install` will run `make -j4`
  # - With `build_jobs: 16` and 32 cores available `spack install` will run `make -j16`
  # - With `build_jobs: 2` and 4 cores available `spack install -j6` will run `make -j6`
  # build_jobs: 16
```

Pascal compute nodes have 36 cores + HT.  Assuming the login nodes are the
same, then `-j 16` seems reasonable enough.  Anyway, if you want to change it,
use `config.yaml` or the command line.

See #26242 and  #11072 and others.

**Comment by adamjstewart** [comment]
I don't think it's possible for Spack to tell whether it's on a login node or compute node or laptop or cloud resource. It's up to the users and sys admins to make sure the defaults are appropriate. We could change Spack to globally default to 1 core when compiling, but that's likely a bad idea for the majority of users.

**Comment by markcmiller86** [comment]
> Are you sure it was 20 and not 16?

Nope. I was looking at all the jobs in `top` output and didn't explicitly count them. Seemed like a lot though!

FYI...I went searching docs for `mac_processors` and when I entered that text on main docs search dialog...it never returned. I waited > 5mins. Oddly though, if I drill down into a specific section of the docs and attempt same search, it returns quickly. 

> I don't think it's possible for Spack to tell whether it's on a login node or compute node or laptop or cloud resource

Ok...I was thinking it might be but maybe not generally. I am just very sensitive to users running parallel jobs on login nodes. Whenever my `vi` takes forever to respond, I invariably see some user running a slew of parallel tasks on the node and a load average above 10. Very annoying so I was hoping for a way for Spack to avoid doing that automagically. Does spack by any chance use `renice` to adjust job priority so that it will use resources if load is low to automatically back of if not?
